execute/plugin.go

```diff
@@ -247,17 +247,20 @@ func (p *Plugin) Observation(
 		return exectypes.NewObservation(groupedCommits, messages, nil).Encode()
 
 	case exectypes.Filter:
-		// TODO: add in nonces, other data comes from previous outcome.
+		// Phase 3: observe nonce for each unique source/sender pair.
 		nonceRequestArgs := make(map[cciptypes.ChainSelector]map[string]struct{})
 
 		// Collect unique senders.
-		for _, commitReport := range previousOutcome.Report.ChainReports {
-			if _, ok := nonceRequestArgs[commitReport.SourceChainSelector]; !ok {
-				nonceRequestArgs[commitReport.SourceChainSelector] = make(map[string]struct{})
+		for _, commitReport := range previousOutcome.PendingCommitReports {
+			if _, ok := nonceRequestArgs[commitReport.SourceChain]; ok {
```

Comment:
Actually, I'm pretty sure this can have multiple reports per source. I'll remove this error case.

It may be worth solving the sorted-map-key issue so that we can improve the types and remove the necessity to validate data structure contents like this.